src/cascadia/TerminalCore/TerminalDispatch.cpp

```diff
@@ -722,3 +725,35 @@ bool TerminalDispatch::HardReset()
 
     return true;
 }
+
+// - ASBSET - Creates and swaps to the alternate screen buffer. In virtual terminals, there exists both a "main"
+//     screen buffer and an alternate. ASBSET creates a new alternate, and switches to it. If there is an already
+//     existing alternate, it is discarded.
+// Arguments:
+// - None
+// Return Value:
+// - True.
+bool TerminalDispatch::UseAlternateScreenBuffer()
+{
+    // TODO GH#3849: When de-duplicating this, the AdaptDispatch version of this calls:
+    // CursorSaveState();
+    _terminalApi.UseAlternateScreenBuffer();
+    _usingAltBuffer = true;
+    return true;
+}
+
+// Routine Description:
+// - ASBRST - From the alternate buffer, returns to the main screen buffer.
+//     From the main screen buffer, does nothing. The alternate is discarded.
+// Arguments:
+// - None
+// Return Value:
+// - True.
+bool TerminalDispatch::UseMainScreenBuffer()
+{
+    _terminalApi.UseMainScreenBuffer();
+    _usingAltBuffer = false;
+    // TODO GH#3849: When de-duplicating this, the AdaptDispatch version of this calls:
+    // CursorRestoreState();

```

Comment:
Again I'm guessing you're relying on conhost handling this, but the VT renderer then needs to take into account that its idea of the last cursor position and last attributes won't necessarily match the terminal on the other end of the conpty pipe.

Reply:
Should be fixed in 54dc304